packages/gatsby/src/schema/extensions/index.js

```diff
@@ -64,6 +64,7 @@ const typeExtensions = {
   },
   nodeInterface: {
     description:
+      `DEPRECATED: Use interface inheritance instead, i.e. "interface Foo implements Node".\n\n` +
```

Comment:
We can't mark directive/extension as depracated?

Reply:
Initially, I've added a deprecation here: https://github.com/gatsbyjs/gatsby/pull/29427/commits/72639a7cbb1df060f6ded517d173927b416652c3 But looks like directives still do not support deprecation, so moved it to the description for visibility.